### ISSUE TITLE
S1172 False positive when pattern matching with parameter (C# 7.7.0.7192)

### ISSUE THREAD

**Issue by tsleegers** [SonarSource/sonar-csharp#2010]
### Description

When a function parameter only is used in het when clause of pattern matching in a switch statement it incorrecty marks that parameter as unused (S1172)

### Repro steps

Please provide the steps required to reproduce the problem

Example code:
```csharp
static int PatternMatch(StringSplitOptions splitOptions, int i)
{
    switch (splitOptions)
    {
        case StringSplitOptions.None
            when i > 0:
            return 1;
        default:
            return 0;
    }
}
```

Parameter _i_ is marked unused.

### Expected behavior

No warning from sonar analyzers.

### Actual behavior

S1172 warning generated:
`
Warning	S1172	Remove this parameter 'i', whose value is ignored in the method.
`

### Known workarounds

Suppress warning.

### Related information

* SonarC# Version: 7.7.0.7192
* Visual Studio Version: Visual Studio Professional 2017 (15.8.7 and 15.9.0 Preview 3)

**Comment by valhristov** [comment]
Thanks for reporting this problem @tsleegers, I managed to reproduce it, we will fix it in a future release.

Note to implementers: for this to be fixed the Live Variable Analysis needs to start analysing patterns.

**Comment by Herdo** [comment]
@valhristov Is there any progress on this issue?

**Comment by Herdo** [comment]
Or @andrei-epure-sonarsource ?

**Comment by andrei-epure-sonarsource** [comment]
@Herdo no progress yet. We plan to invest some effort this year in improving our Control Flow Graph.
